README.md

```diff
@@ -464,10 +464,77 @@ progress = (
 </tr>
 </table>
 
+#### Regular Expressions
+
+It can be useful to sort an entire group based on a non-prefix substring. The
+option `by_regex=…` takes a comma-separated list of [regular
+expressions] that will be applied to the group, and then sorting
+will take place on just the results of the regular expressions.
+
+> [!TIP]
+> Regular expressions often need special characters. See [Syntax](#syntax) below
+> for how to include special characters in the `by_regex` option.
+
+By default, all characters that the regular expression matches will be
+considered for sorting. If the regular expression contains any capturing groups,
+only the characters matched by the capturing groups will be considered for
+sorting. The result from each regular expression will be concatenated into a
+list of results, and that list of results will be sorted [lexicographically].
```

Comment:
Just curious, do we allow users to use `by_regex` with `numeric`? Probably out of the scope of this change though.